zerostash-fuse/src/chunks.rs

```diff
@@ -0,0 +1,140 @@
+use infinitree::{
+    object::{AEADReader, PoolRef, Reader},
+    ChunkPointer,
+};
+use std::{iter::Skip, sync::Arc, vec::IntoIter};
+
+type Chunks = Skip<IntoIter<(u64, Arc<ChunkPointer>)>>;
+
+pub struct ChunksIter {
+    pub chunks: std::iter::Peekable<Chunks>,
+}
+
+impl ChunksIter {
+    fn new(chunks: Chunks) -> Self {
+        let chunks = chunks.peekable();
+        Self { chunks }
+    }
+
+    fn peek_next_offset(&mut self, file_size: usize) -> usize {
+        let arc = (file_size as u64, Arc::new(ChunkPointer::default()));
+        let (chunk_offset, _) = self.chunks.peek().unwrap_or(&arc);
+        *chunk_offset as usize
+    }
+
+    fn get_next(&mut self) -> Option<(usize, Arc<ChunkPointer>)> {
+        let (c_offset, pointer) = match self.chunks.next() {
```

Comment:
You could drop the `let` and just simply keep the body of `match` as the function body, as it's an expression. You then also don't need the `return None`, and just say `None => None`.